src/webgpu/api/operation/rendering/basic.spec.ts

```diff
@@ -129,4 +137,191 @@ g.test('large_draw')
+      params[4] = 0; // First instance (indexed)
+      t.device.queue.writeBuffer(indirectBuffer, 0, params, 0, 5);
+    };
+
+    const MILLION = 1024 * 1024;
+    const MAX_INDICES = 16 * MILLION;
+    const indexBuffer = t.device.createBuffer({
+      size: MAX_INDICES * Uint32Array.BYTES_PER_ELEMENT,
+      usage: GPUBufferUsage.INDEX | GPUBufferUsage.COPY_DST,
+    });
+    const indexData = new Uint32Array(MILLION);
+    for (let offset = 0; offset < MAX_INDICES; offset += MILLION) {
+      for (let k = 0; k < MILLION; ++k) {
+        indexData[k] = offset + k;
+      }
+      t.device.queue.writeBuffer(
+        indexBuffer,
+        offset * Uint32Array.BYTES_PER_ELEMENT,
+        indexData,
+        0,
+        MILLION
+      );
+    }
+
+    const colorAttachment = t.device.createTexture({
+      format: 'rgba8unorm',
+      size: { width: 3, height: 3, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+    const colorAttachmentView = colorAttachment.createView();
+
+    const runPipeline = async (numInstances: number, numVertices: number) => {
+      const pipeline = t.device.createRenderPipeline({
+        vertex: {
+          module: t.device.createShaderModule({
+            code: `
+            struct Output {
+              [[builtin(position)]] position: vec4<f32>;
+              [[location(0)]] color: vec4<f32>;
+            };
+
+            fn selectValue(index: u32, maxIndex: u32) -> f32 {
+              let highOrMid = select(1.0, 0.5, index == maxIndex - 1u);
+              return select(0.0, highOrMid, index == 0u);
+            }
+
+            [[stage(vertex)]] fn main(
+                [[builtin(vertex_index)]] v: u32,
+                [[builtin(instance_index)]] i: u32)
+                -> Output {
+              let r = selectValue(v, ${numVertices}u);
+              let b = selectValue(i, ${numInstances}u);
+              let x = (r * 4.0 - 2.0) / 3.0;
+              let y = (b * 4.0 - 2.0) / -3.0;
+              return Output(vec4<f32>(x, y, 0.0, 1.0),
+                            vec4<f32>(r, 0.0, b, 1.0));
+            }
+            `,
+          }),
+          entryPoint: 'main',
+        },
+        fragment: {
+          module: t.device.createShaderModule({
+            code: `
+              [[stage(fragment)]] fn main([[location(0)]] color: vec4<f32>)
+                  -> [[location(0)]] vec4<f32> {
+                return color;
+              }
+              `,
+          }),
+          entryPoint: 'main',
+          targets: [{ format: 'rgba8unorm' }],
+        },
+        primitive: { topology: 'point-list' },
+      });
+
+      const encoder = t.device.createCommandEncoder();
+      const pass = encoder.beginRenderPass({
+        colorAttachments: [
+          {
+            view: colorAttachmentView,
+            storeOp: 'store',
+            loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
```

Comment:
This will be the "background" pixel color of any pixel that doesn't get a point drawn to it, but I think it's the same color as some of the points. Maybe change it to another color like (0,1,0,1)?

Reply:
Made it blue, and shader output yellow